ex1/dates_times.py

```diff
@@ -26,19 +26,85 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
+from datetime import date, timedelta, time
+
 '''
 Exercises for using the datetime and the calendar module
 '''
 
+
 # Define a function named last_of_month that takes an argument dt of type date
 # and returns a date object representing the last day of the month dt was in.
+def last_of_month(dt):
+    """
+    Returns the last day of a month.
+
+    Parameters
+    ----------
+    dt : datetime
+
+    Returns
+    -------
+    lastday : datetime
+    """
+    # jump to the first day of the next month and subtract one day
+    if dt.month == 12:
+        m = 1
+        y = dt.year + 1
+    else:
+        m = dt.month + 1
+        y = dt.year
+    lastday = date(y, m, 1) + timedelta(days=-1)
+    return lastday
```

Comment:
The [calendar.monthrange](https://docs.python.org/3.4/library/calendar.html#calendar.monthrange) function would have made your live easier.